### ISSUE TITLE
TF 1.15.3 Object Detection API Not Reading "fine_tune_checkpoint" from config

### ISSUE THREAD

**Issue by rohan-pradhan** [tensorflow/models#9384]
# Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [ ] I am using the latest TensorFlow Model Garden release and TensorFlow 2.
- [X] I am reporting the issue to the correct repository. (Model Garden official or research directory)
- [X] I checked to make sure that this issue has not been filed already.

## 1. The entire URL of the file you are using

https://github.com/tensorflow/models/blob/master/research/object_detection/samples/configs/ssd_mobilenet_v2_coco.config 

https://github.com/tensorflow/models/blob/master/research/object_detection/model_main.py

## 2. Describe the bug

When I run the model_main.py to train my SSD model, the pipeline seems to be skipping the fine_tune_checkpoint information from the config file. Therefore, the model initializes the weight from scratch (I believe). 

## 3. Steps to reproduce

1. Download pretrained model from TF 1.0 Object Detection Zoo 
2. Uncompress file 
3. Move config file to appropriate location 
4. Make edits to the config file (change # of classes and add appropriate paths) 
5. Run `python model_main.py --model_dir models/ssd_mobilenet_v2_coco/ --pipeline_config_path models/ssd_mobilenet_v2_coco/pipeline.config `

## 4. Expected behavior
I expected that upon startup, the program would restore from the fine_tune_checkpoint. However, it does not seem to be doing that. 

## 5. Additional context

When I run the model_main.py I get this output: 
```Call initializer instance with the dtype argument instead of passing it to the constructor
INFO:tensorflow:Done calling model_fn.
I1016 22:57:11.198012 139805132744512 estimator.py:1150] Done calling model_fn.
INFO:tensorflow:Create CheckpointSaverHook.
I1016 22:57:11.199217 139805132744512 basic_session_run_hooks.py:541] Create CheckpointSaverHook.
INFO:tensorflow:Graph was finalized.
I1016 22:57:14.707384 139805132744512 monitored_session.py:240] Graph was finalized.
INFO:tensorflow:Running local_init_op.
I1016 22:57:21.371568 139805132744512 session_manager.py:500] Running local_init_op.
INFO:tensorflow:Done running local_init_op.
I1016 22:57:21.885310 139805132744512 session_manager.py:502] Done running local_init_op. 
```

It is my understanding that there should be a line for "Restoring parameters..." between Graph was finalized and the running local_init_op. On top of that, when I deliberately  give a wrong path to the fine_tune_checkpoint in the config file, no error is given. 

Here is the current config file I am using: 
```
model {
  ssd {
    num_classes: 2
    image_resizer {
      fixed_shape_resizer {
        height: 300
        width: 300
      }
    }
    feature_extractor {
      type: "ssd_mobilenet_v2"
      depth_multiplier: 1.0
      min_depth: 16
      conv_hyperparams {
        regularizer {
          l2_regularizer {
            weight: 3.99999989895e-05
          }
        }
        initializer {
          truncated_normal_initializer {
            mean: 0.0
            stddev: 0.0299999993294
          }
        }
        activation: RELU_6
        batch_norm {
          decay: 0.999700009823
          center: true
          scale: true
          epsilon: 0.0010000000475
          train: true
        }
      }
      
      use_depthwise: true
    }
    box_coder {
      faster_rcnn_box_coder {
        y_scale: 10.0
        x_scale: 10.0
        height_scale: 5.0
        width_scale: 5.0
      }
    }
    matcher {
      argmax_matcher {
        matched_threshold: 0.5
        unmatched_threshold: 0.5
        ignore_thresholds: false
        negatives_lower_than_unmatched: true
        force_match_for_each_row: true
      }
    }
    similarity_calculator {
      iou_similarity {
      }
    }
    box_predictor {
      convolutional_box_predictor {
        conv_hyperparams {
          regularizer {
            l2_regularizer {
              weight: 3.99999989895e-05
            }
          }
          initializer {
            truncated_normal_initializer {
              mean: 0.0
              stddev: 0.0299999993294
            }
          }
          activation: RELU_6
          batch_norm {
            decay: 0.999700009823
            center: true
            scale: true
            epsilon: 0.0010000000475
            train: true
          }
        }
        min_depth: 0
        max_depth: 0
        num_layers_before_predictor: 0
        use_dropout: false
        dropout_keep_probability: 0.800000011921
        kernel_size: 3
        box_code_size: 4
        apply_sigmoid_to_scores: false
      }
    }
    anchor_generator {
      ssd_anchor_generator {
        num_layers: 6
        min_scale: 0.20000000298
        max_scale: 0.949999988079
        aspect_ratios: 1.0
        aspect_ratios: 2.0
        aspect_ratios: 0.5
        aspect_ratios: 3.0
        aspect_ratios: 0.333299994469
      }
    }
    post_processing {
      batch_non_max_suppression {
        score_threshold: 0.300000011921
        iou_threshold: 0.600000023842
        max_detections_per_class: 100
        max_total_detections: 100
      }
      score_converter: SIGMOID
    }
    normalize_loss_by_num_matches: true
    loss {
      localization_loss {
        weighted_smooth_l1 {
        }
      }
      classification_loss {
        weighted_sigmoid {
        }
      }
      hard_example_miner {
        num_hard_examples: 3000
        iou_threshold: 0.990000009537
        loss_type: CLASSIFICATION
        max_negatives_per_positive: 3
        min_negatives_per_image: 3
      }
      classification_weight: 1.0
      localization_weight: 1.0
    }
  }
}
train_config {
  batch_size: 24
  data_augmentation_options {
    random_horizontal_flip {
    }
  }
  data_augmentation_options {
    ssd_random_crop {
    }
  }
  optimizer {
    rms_prop_optimizer {
      learning_rate {
        exponential_decay_learning_rate {
          initial_learning_rate: 0.00400000018999
          decay_steps: 800720
          decay_factor: 0.949999988079
        }
      }
      momentum_optimizer_value: 0.899999976158
      decay: 0.899999976158
      epsilon: 1.0
    }
  }
  fine_tune_checkpoint: "pre-trained-models/ssd_mobilenet_v2_coco_2018_03_29/checkpoint"
  from_detection_checkpoint: true
  load_all_detection_checkpoint_vars: true
  num_steps: 200000
  fine_tune_checkpoint_type: "detection"
}
train_input_reader {
  label_map_path: "data/labelmap.pbtxt"
  tf_record_input_reader {
    input_path: "data/train_dataset.record-00000-of-00010"
    input_path: "data/train_dataset.record-00001-of-00010"
    input_path: "data/train_dataset.record-00002-of-00010"
    input_path: "data/train_dataset.record-00003-of-00010"
    input_path: "data/train_dataset.record-00004-of-00010"
    input_path: "data/train_dataset.record-00005-of-00010"
    input_path: "data/train_dataset.record-00006-of-00010"
    input_path: "data/train_dataset.record-00007-of-00010"
    input_path: "data/train_dataset.record-00008-of-00010"
    input_path: "data/train_dataset.record-00009-of-00010"
  }
}
eval_config {
  num_examples: 2000
  metrics_set: "coco_detection_metrics"
  use_moving_averages: false
}
eval_input_reader {
  label_map_path: "data/labelmap.pbtxt"
  shuffle: false
  num_readers: 1
  tf_record_input_reader {
    input_path: "data/val_dataset.record-00000-of-00001"
  }
}
```
In terms of the fine_tune_checkpoint path, I have tried both the checkpoint file and the model.ckpt paths. Neither made a difference. 

The training process technically works, but I can tell that it isn't restoring from the pre-trained checkpoint as the loss starts off very high (30+) and then stabilizes at 10-12 and does not improve from then. 

## 6. System information
- AWS Linux Ubuntu 18.04
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
- Mobile device name if the issue happens on a mobile device:
- Tensorflow Version 1.15.3 (pre-installed via AWS Deep Learning AMI)

**Comment by ravikyram** [comment]
@rohan-pradhan 

Will it be possible to share colab link to reproduce the issue in our environment.It helps us in localizing the issue faster.Thanks!
